pkg/cosign/keys.go

```diff
@@ -51,45 +60,82 @@ func GeneratePrivateKey() (*ecdsa.PrivateKey, error) {
 	return ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
 }
 
-func GenerateKeyPair(pf PassFunc) (*Keys, error) {
-	priv, err := GeneratePrivateKey()
+func ImportKeyPair(keyPath string, pf PassFunc) (*KeysBytes, error) {
+
+	kb, err := os.ReadFile(filepath.Clean(keyPath))
 	if err != nil {
 		return nil, err
 	}
 
-	x509Encoded, err := x509.MarshalPKCS8PrivateKey(priv)
+	p, _ := pem.Decode(kb)
+	if p == nil {
+		return nil, errors.New("invalid pem block")
+	}
+
+	switch p.Type {
+
+	case RSAPrivateKeyPemType:
+		pk, err := x509.ParsePKCS1PrivateKey(p.Bytes)
+		if err != nil {
+			return nil, fmt.Errorf("parsing error")
+		}
+		return marshalKeyPair(Keys{pk, pk.Public()}, pf)
```

Comment:
move this line under the switch

Reply:
not sure I follow here.  Can you explain?

Reply:
Since this line's repeated in both cases, it can just occur after the switch statement

```
switch p.Type {
	case RSAPrivateKeyPemType:
		pk, err := x509.ParsePKCS1PrivateKey(p.Bytes)
		if err != nil {
			return nil, fmt.Errorf("parsing error")
		}
	default:
		pk, err := x509.ParseECPrivateKey(p.Bytes)
		if err != nil {
			return nil, fmt.Errorf("parsing error")
		}
	}
        return marshalKeyPair(Keys{pk, pk.Public()}, pf)
}
```